app/parse.py

```diff
@@ -1,12 +1,132 @@
-from dataclasses import dataclass
+import csv
+import time
+
+import requests
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from dataclasses import dataclass, fields
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://mate.academy"
+
+
+class Driver:
+    Instance = None
```

Comment:
It's not a good practice to use Singleton pattern in Python because it can lead to unexpected behavior due to its global state. Consider using a factory function or a class method instead.